amgut/lib/data_access/test/test_ag_data_access.py

```diff
@@ -64,13 +67,162 @@ def test_getAnimalParticipants(self):
+                'site_sampled': 'Stool'},
+               {'status': 'Received',
+                'sample_time': datetime.time(9, 50),
+                'notes': 'REMOVED',
+                'barcode': '000004214',
+                'sample_date': datetime.date(2013, 10, 14),
+                'site_sampled': 'Stool'},
+               {'status': 'Received',
+                'sample_time': datetime.time(12, 0),
+                'notes': 'REMOVED',
+                'barcode': '000004215',
+                'sample_date': datetime.date(2013, 10, 13),
+                'site_sampled': 'Stool'},
+               {'status': 'Received',
+                'sample_time': datetime.time(9, 30),
+                'notes': 'REMOVED',
+                'barcode': '000004216',
+                'sample_date': datetime.date(2013, 10, 15),
+                'site_sampled': 'Stool'},
+               {'status': 'Received',
+                'sample_time': datetime.time(14, 25),
+                'notes': 'REMOVED',
+                'barcode': '000004218',
+                'sample_date': datetime.date(2013, 10, 12),
+                'site_sampled': 'Stool'},
+               {'status': 'Received',
+                'sample_time': datetime.time(10, 15),
+                'notes': 'REMOVED',
+                'barcode': '000004219',
+                'sample_date': datetime.date(2013, 10, 17),
+                'site_sampled': 'Stool'}]
+        self.assertItemsEqual(res, exp)
+
+    def test_getParticipantSamplesNotPresent(self):
+        i = '00000000-0000-0000-0000-000000000000'
+        res = self.ag_data.getParticipantSamples(i, "REMOVED")
+        self.assertEqual(res, [])
 
     def test_getEnvironmentalSamples(self):
-        raise NotImplementedError()
+        i = "d6b0f287-b9d9-40d4-82fd-a8fd3db6c476"
+        res = self.ag_data.getEnvironmentalSamples(i)
+        exp = [{'status': None, 'sample_time': datetime.time(21, 45),
+                'notes': 'REMOVED', 'barcode': '000028433',
+                'sample_date': datetime.date(2015, 6, 7),
+                'site_sampled': None}]
+        self.assertItemsEqual(res, exp)
+
+    def test_getEnvironmentalSamplesNotPresent(self):
+        i = '00000000-0000-0000-0000-000000000000'
+        res = self.ag_data.getEnvironmentalSamples(i)
+        self.assertEqual(res, [])
 
     def test_getAvailableBarcodes(self):
-        raise NotImplementedError()
+        i = "d8592c74-9694-2135-e040-8a80115d6401"
+        res = self.ag_data.getAvailableBarcodes(i)
+        exp = ['000005628', '000005627', '000005624',
+               '000005625', '000005626', '000004217']
+        self.assertItemsEqual(res, exp)
+
+        i = "d6b0f287-b9d9-40d4-82fd-a8fd3db6c476"
+        res = self.ag_data.getAvailableBarcodes(i)
+        exp = ['000028434']
+        self.assertItemsEqual(res, exp)
+
+    def test_getAvailableBarcodesNotPresent(self):
+        i = '00000000-0000-0000-0000-000000000000'
+        res = self.ag_data.getAvailableBarcodes(i)
```

Comment:
Should it raise?

On Sun, Sep 27, 2015 at 4:59 PM, mortonjt <notifications@github.com> wrote:

> In amgut/lib/data_access/test/test_ag_data_access.py
> <https://github.com/biocore/american-gut-web/pull/500#discussion_r40511029>
> :
>
> >
> >      def test_getAvailableBarcodes(self):
> > -        raise NotImplementedError()
> > +        i = "d8592c74-9694-2135-e040-8a80115d6401"
> > +        res = self.ag_data.getAvailableBarcodes(i)
> > +        exp = ['000005628', '000005627', '000005624',
> > +               '000005625', '000005626', '000004217']
> > +        self.assertItemsEqual(res, exp)
> > +
> > +        i = "d6b0f287-b9d9-40d4-82fd-a8fd3db6c476"
> > +        res = self.ag_data.getAvailableBarcodes(i)
> > +        exp = ['000028434']
> > +        self.assertItemsEqual(res, exp)
> > +
> > +    def test_getAvailableBarcodesNotPresent(self):
> > +        i = '00000000-0000-0000-0000-000000000000'
> > +        res = self.ag_data.getAvailableBarcodes(i)
>
> If it isn't in the db, what would be an appropriate way to handle such a
> query?
>
> Right now, it is either returning None or []
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/american-gut-web/pull/500/files#r40511029>.
>